SAT/Z3Interfacing.cpp

```diff
@@ -98,50 +98,63 @@ const char* errToString(Z3_error_code code)
   }
 }
 
-void handleZ3Error(Z3_context ctxt, Z3_error_code code) 
+void handleZ3Error(Z3_context ctxt, Z3_error_code code)
 {
   DEBUG(errToString(code))
   throw z3::exception(errToString(code));
 }
 
 #define STATEMENTS_TO_EXPRESSION(...) [&]() { __VA_ARGS__; return 0; }()
 
-Z3Interfacing::Z3Interfacing(SAT2FO& s2f, bool showZ3, bool unsatCore):
-  _varCnt(0), 
+Z3Interfacing::Z3Interfacing(SAT2FO& s2f, bool showZ3, bool unsatCore, vstring const& exportSmtlib):
+  _varCnt(0),
   _sat2fo(s2f),
-  _status(SATISFIABLE), 
+  _status(SATISFIABLE),
   _config(),
   _context(_config),
   _solver(_context),
   _model((STATEMENTS_TO_EXPRESSION(
-            BYPASSING_ALLOCATOR; 
-            _solver.check(); 
+            BYPASSING_ALLOCATOR;
+            _solver.check();
           ),
-         _solver.get_model())), 
+         _solver.get_model())),
   _assumptions(),
   _showZ3(showZ3),
-  _unsatCore(unsatCore)
+  _unsatCore(unsatCore),
+  _out(exportSmtlib == "" ? Option<std::ofstream>()
+                          : Option<std::ofstream>(std::ofstream(exportSmtlib.c_str())) )
 {
   CALL("Z3Interfacing::Z3Interfacing");
   BYPASSING_ALLOCATOR
   _solver.reset();
 
-  z3::set_param("rewriter.expand_store_eq", "true");
+  if (_out.isSome() && _out.unwrap().fail()) {
+    throw UserErrorException("Failed to open file: ", exportSmtlib);
+  }
+  outputln("(check-sat)");
+  outputln("(get-model)");
+  outputln("(reset)");
```

Comment:
Vampire itself doesn't handle reset...  although I started to add in a branch a long time ago. But I'm guessing the aim is to hand these SMT-LIB files to other solvers!

Reply:
Yes so the idea is to export our z3 calls in an smtlib file in order to b able to file issues for the z3 people easily.

Reply:
I think it could help with performance things as well as debugging. I'd also like to use this interface to collect some problems and submit them to the interactive part of SMT-LIB